tests/ai/test_databricks_client.py

```diff
@@ -262,6 +264,37 @@ def test_create_and_execute_function(
             assert result.value == function_sample.output
 
 
+@requires_databricks
+def test_execute_function_with_timeout(client: DatabricksFunctionClient, monkeypatch):
+    monkeypatch.setenv(UNITYCATALOG_AI_CLIENT_EXECUTION_TIMEOUT, "5")
+    with generate_func_name_and_cleanup(client) as func_name:
+        full_func_name = f"{CATALOG}.{SCHEMA}.{func_name}"
+        sql_body = f"""CREATE FUNCTION {full_func_name}()
+RETURNS STRING
+LANGUAGE PYTHON
+AS $$
+    import time
+
+    time.sleep(100)
+    return "10"
+$$
+"""
+        client.create_function(sql_function_body=sql_body)
+        time1 = time.time()
+        result = client.execute_function(full_func_name)
+        time_total1 = time.time() - time1
+        assert result.error.startswith("Statement execution is still pending after 5 seconds")
+
+        time2 = time.time()
+        result = client.execute_function(
+            full_func_name, {EXECUTE_FUNCTION_ARG_NAME: {"wait_timeout": "10s"}}
+        )
+        time_total2 = time.time() - time2
+        assert result.error.startswith("Statement execution is still pending after 5 seconds")
+        # default wait_timeout is 30s
+        assert abs(abs(time_total2 - time_total1) - 20) < 5
```

Comment:
where does 20 come from?

Reply:
default wait_timeout in `DEFAULT_EXECUTE_FUNCTION_ARGS` passed to `self.client.statement_execution.execute_statement` is 30s, and the second test we override that to 10s, 30-10=20

Reply:
do we need this line?

Reply:
Yea I want to make sure the parameters passed as EXECUTE_FUNCTION_ARG_NAME works, as part of the integration test. But I can split it to a different test if you think this is too much?

Reply:
Let's split it up.